### PR TITLE
Remove hardcoded NameList/EDMF parameters

### PR DESCRIPTION
**Pull request by nefrathenrici**
## Purpose 
This implements #1305 

This PR removes all namelist functionality, replacing it with calls to ClimaParameters. This includes the whole `generate_namelist ` file and all usage of `parse_namelist`. 

The existing `TurbulenceConvectionParameters` struct has been expanded to include all former namelist parameters. This includes a few microphysics parameters as well. In most cases where `namelist` is passed in, I have just replaced it with the expanded struct.

I also changed the EDMF case parsing, passing the case name into `tc_parameter_set`.

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/TurbulenceConvection/Parameters.jl [hunk 1]
```diff
@@ -23,6 +23,44 @@ Base.@kwdef struct TurbulenceConvectionParameters{FT, MP, SFP} <: ATCP
     microph_scaling_melt::FT # TODO: move to microphysics parameter set? or Microphysics1Moment?
     microphys_params::MP
     surf_flux_params::SFP
+    case_name::String
```
Comment (charleskawczynski) [inline src/TurbulenceConvection/Parameters.jl:26]: Let's remove the strings since they're GPU-incompatible. We can use types instead, and that way we can also leverage dispatch.

Also, we should use `::Int` over `::Integer` since `Integer` is an abstract type and we should avoid structs with abstract types (see https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-fields-with-abstract-type)

File: examples/Manifest.toml [hunk 2]
```diff
@@ -157,9 +157,9 @@ version = "0.1.2"
 
 [[deps.CLIMAParameters]]
 deps = ["DocStringExtensions", "TOML", "Test"]
-git-tree-sha1 = "8cb1d69b6b5fc41be48bea969a68763a17843f1e"
+git-tree-sha1 = "a769996f7ed7a2f1fb070c4837370603387e5eb6"
```
Comment (charleskawczynski) [inline examples/Manifest.toml:160]: We can make these changes in a separate `julia .dev/up_deps.jl` PR, since changes miss the other environments (e.g., docs, test, perf)

Comment (charleskawczynski) [reply examples/Manifest.toml:160]: I'll open one now

Comment (charleskawczynski) [reply examples/Manifest.toml:160]: Opened #1335

File: src/TurbulenceConvection/Parameters.jl (hunk 1 above)
Comment (charleskawczynski) [reply src/TurbulenceConvection/Parameters.jl:26]: Also, `case` is not needed, nor is it currently used, past initialization, and we don't want simulations to be case-dependent (this is why _all_ case information in SCAMPy was in a `Cases` file). We should avoid making it available in our RHS functions to ensure it doesn't happen.

Comment (nefrathenrici) [reply src/TurbulenceConvection/Parameters.jl:26]: I can turn the String fields into types, and fix the Integer fields.
Should I get rid of the `case_name `and `case ` occurrences entirely then?

File: examples/Manifest.toml (hunk 2 above)
Comment (trontrytel) [reply examples/Manifest.toml:160]: The above PR got merged, So we should be able to rebase this one

File: src/TurbulenceConvection/Parameters.jl (hunk 1 above)
Comment (charleskawczynski) [reply src/TurbulenceConvection/Parameters.jl:26]: It looks like it's being deleted in #1327, maybe the simplest thing for now is to grab `case_name` from `parsed_args` (which is available in `turbconv_cache`). Do we need to do anything with `case`?

Comment (nefrathenrici) [reply src/TurbulenceConvection/Parameters.jl:26]: I don't think we do, I couldn't find any examples of it actually being used.

File: src/TurbulenceConvection/Parameters.jl [hunk 3]
```diff
@@ -8,22 +8,21 @@ import Thermodynamics as TD
 import SurfaceFluxes as SF
 import CloudMicrophysics as CM
 
+include("../model_getters.jl")
```
Comment (charleskawczynski) [inline src/TurbulenceConvection/Parameters.jl:11]: We probably shouldn't re-include this, why is this necessary?

Comment (charleskawczynski) [reply src/TurbulenceConvection/Parameters.jl:11]: (same as below)

File: src/TurbulenceConvection/types.jl [hunk 4]
```diff
@@ -1,3 +1,5 @@
+include("../model_getters.jl")
+import ClimaAtmos.TurbulenceConvection.Parameters as TCP
```
Comment (charleskawczynski) [inline src/TurbulenceConvection/types.jl:2]:
```suggestion
import .Parameters as TCP
```
I think this should work, we shouldn't ever need / use `import ClimaAtmos.anything` inside `src/` of `ClimaAtmos`. It technically opens the door to use multiple versions of ClimaAtmos.

Comment (charleskawczynski) [reply src/TurbulenceConvection/types.jl:2]: (same elsewhere)

File: src/model_getters.jl [hunk 5]
```diff
@@ -359,3 +359,59 @@ function ThermoDispatcher(atmos)
...
+    elseif precip_fraction_model_name == "cloud_cover"
+        return DiagnosticPrecipFraction(paramset.precip_fraction_limiter)
+    else
+        error(
+            "Something went wrong. Invalid `precip_fraction` model: `$precip_fraction_model_name`",
+        )
+    end
+end
+
+function AbstractQuadratureType(s::String)
```
Comment (charleskawczynski) [inline src/model_getters.jl:390]: This is a confusing name, because Julia prohibits creating instances of abstract types, can we just call it `QuadratureType`?

Comment (charleskawczynski) [reply src/model_getters.jl:390]: Also, why not just pass in a Dict and read it from the Dict? Can we also error if none of the cases are satisfied?

Comment (nefrathenrici) [reply src/model_getters.jl:390]: Maybe it makes sense to just move it all into the TurbulenceConvectionParameters? It shouldn't get used anywhere else.
